src/hyperlight_host/src/hypervisor/hyperlight_vm.rs

```diff
@@ -0,0 +1,1075 @@
+        }
+
+        // Mark initial setup as complete for Windows - subsequent map_memory calls will fail
+        #[cfg(target_os = "windows")]
+        vm.complete_initial_memory_setup();
+
+        #[cfg(feature = "init-paging")]
+        vm.set_sregs(&CommonSpecialRegisters::standard_64bit_defaults(_pml4_addr))?;
+        #[cfg(not(feature = "init-paging"))]
+        vm.set_sregs(&CommonSpecialRegisters::standard_real_mode_defaults())?;
+
+        #[cfg(gdb)]
+        let gdb_conn = if let Some(gdb_conn) = gdb_conn {
+            // Add breakpoint to the entry point address
+            vm.set_debug(true)?;
+            vm.add_hw_breakpoint(entrypoint)?;
+
+            Some(gdb_conn)
+        } else {
+            None
+        };
+
+        let rsp_gp = GuestPtr::try_from(RawPtr::from(rsp))?;
+
+        #[cfg(any(kvm, mshv3))]
+        let interrupt_handle: Arc<dyn InterruptHandleImpl> = Arc::new(LinuxInterruptHandle {
+            state: AtomicU8::new(0),
+            #[cfg(all(
+                target_arch = "x86_64",
+                target_vendor = "unknown",
+                target_os = "linux",
+                target_env = "musl"
+            ))]
+            tid: AtomicU64::new(unsafe { libc::pthread_self() as u64 }),
+            #[cfg(not(all(
+                target_arch = "x86_64",
+                target_vendor = "unknown",
+                target_os = "linux",
+                target_env = "musl"
+            )))]
+            tid: AtomicU64::new(unsafe { libc::pthread_self() }),
+            retry_delay: config.get_interrupt_retry_delay(),
+            sig_rt_min_offset: config.get_interrupt_vcpu_sigrtmin_offset(),
+            dropped: AtomicBool::new(false),
+        });
+
+        #[cfg(target_os = "windows")]
+        let interrupt_handle: Arc<dyn InterruptHandleImpl> = Arc::new(WindowsInterruptHandle {
+            state: AtomicU8::new(0),
+            partition_handle: vm.partition_handle(),
+            dropped: AtomicBool::new(false),
+        });
+
+        #[cfg_attr(not(gdb), allow(unused_mut))]
+        let mut ret = Self {
+            vm,
+            entrypoint,
+            orig_rsp: rsp_gp,
+            interrupt_handle,
+            page_size: 0, // Will be set in `initialise`
+
+            next_slot: mem_regions.len() as u32,
+            sandbox_regions: mem_regions,
+            mmap_regions: Vec::new(),
+            freed_slots: Vec::new(),
+
+            #[cfg(gdb)]
+            gdb_conn,
+            #[cfg(gdb)]
+            sw_breakpoints: HashMap::new(),
+            #[cfg(feature = "mem_profile")]
+            trace_info,
+            #[cfg(crashdump)]
+            rt_cfg,
+        };
+
+        // Send the interrupt handle to the GDB thread if debugging is enabled
+        // This is used to allow the GDB thread to stop the vCPU
+        #[cfg(gdb)]
+        if ret.gdb_conn.is_some() {
+            ret.send_dbg_msg(DebugResponse::InterruptHandle(ret.interrupt_handle.clone()))?;
+        }
```

Comment:
if you move the `interrupt_handle` creation a bit earlier, then you can do this at the same time that you do ` vm.set_debug(true)?;`